api/base/serializers.py

```diff
@@ -143,92 +173,141 @@ def get_attribute(self, obj):
         return field_source_method(auth=auth)
 
 
-class JSONAPIHyperlinkedIdentityField(ser.HyperlinkedIdentityField):
+class RelationshipField(ser.HyperlinkedIdentityField):
     """
-    HyperlinkedIdentityField that returns a nested dict with url,
-    optional meta information, and link_type.
+    RelationshipField that permits the return of both self and related links, along with optional
+    meta information.
 
     Example:
+    children = RelationshipField(
+        related_view='nodes:node-children',
+        related_view_kwargs={'node_id': 'pk'},
+        self_view='nodes:node-node-children-relationship',
+        self_view_kwargs={'node_id': 'pk'},
+        related_meta={'count': 'get_node_count'}
+    )
 
-        children = JSONAPIHyperlinkedIdentityField(view_name='nodes:node-children', lookup_field='pk',
-                                    link_type='related', lookup_url_kwarg='node_id', meta={'count': 'get_node_count'})
+    If you surround the lookup_field in angular brackets, it will not attempt to find that attribute on the target, but rather
```

Comment:
Let me clarify my reasoning. As an outside developer reading the following code,

```python
wiki_home = RelationshipField(
    related_view='addon:addon-detail',
    related_view_kwargs={'node_id': '<_id>', 'provider': 'wiki'},
)
```

I would intuit that `reverse('addon:addon-detail, provider='wiki', ...)` was being used under the hood, but `<_id>` was passed in some special way. The documentation would tell me that `<>` means attribute access.

Consider a URL that wouldn't need to pull an attribute:

```python
wiki_summary = RelationshipField(
    related_view='addon:addon-summary',
    related_view_kwargs={'provider': 'wiki'},
)
```

As an outside developer, would you expect that (A) `provider='wiki'` was being passed to `reverse()` OR (B) `provider=node.wiki` was being passed to `reverse()`? I would expect (A) to be the most common answer.

In sum, the `<>` syntax is an acknowledgement that there is some "magic" going on to get the argument.

Realizing that this is largely personal preference, pinging @brianjgeiger and/or @cwisecarver for a tiebreak =)